transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringCompletionQueue.java

```diff
@@ -51,37 +54,44 @@
     this.ringSize = ringSize;
     this.ringAddress = ringAddress;
     this.ringFd = ringFd;
+
+    this.ringMask = PlatformDependent.getInt(kringMaskAddress);
+    this.ringHead = PlatformDependent.getInt(kHeadAddress);
+  }
+
+  public boolean hasCompletions() {
+      return ringHead != PlatformDependent.getIntVolatile(kTailAddress);
   }
 
   public int process(IOUringCompletionQueueCallback callback) {
+      int tail = PlatformDependent.getIntVolatile(kTailAddress);
```

Comment:
It looks correct that we are getting the completion queue tail with volatile, but previously the volatile load was on the head. Can we have a double-check and justify to ourselves that this is right?

Reply:
I think this is right... agree it would be good for others to confirm :)

Reply:
@nitsanw can you have a look as well to ensure that we are ok in terms with memory barriers etc ?